src/components/axes/categoryAxis.ts

```diff
@@ -89,19 +108,34 @@ export module Axes {
         throw new Error("Angle " + angle + " not supported; only 0, 90, and -90 are valid values");
       }
       this._tickLabelAngle = angle;
-      this._invalidateLayout();
+      this.invalidateLayout();
       return this;
     }
 
+    protected getTickValues(): string[] {
+      return this.scale.domain();
+    }
+
+    protected rescale() {
+      return this.invalidateLayout();
+    }
+
+    protected setup() {
+      super.setup();
+      this.measurer = new SVGTypewriter.Measurers.CacheCharacterMeasurer(this.tickLabelContainer);
+      this.wrapper = new SVGTypewriter.Wrappers.SingleLineWrapper();
+      this.writer = new SVGTypewriter.Writers.Writer(this.measurer, this.wrapper);
+    }
+
     /**
      * Measures the size of the ticks while also writing them to the DOM.
      * @param {D3.Selection} ticks The tick elements to be written to.
      */
-    private _drawTicks(axisWidth: number, axisHeight: number, scale: Scales.Category, ticks: D3.Selection) {
+    private drawTicks(axisWidth: number, axisHeight: number, scale: Scales.Category, ticks: D3.Selection) {
       var self = this;
       var xAlign: {[s: string]: string};
       var yAlign: {[s: string]: string};
-      switch (this.tickLabelAngle()) {
+      switch (this._tickLabelAngle) {
```

Comment:
What's the difference? This one takes advantage of the private variable that can be accessed in the class; no reason to use the public getter method.